rclrs/src/logging/logger.rs

```diff
@@ -0,0 +1,101 @@
+use std::{borrow::Borrow, ffi::CString};
+
+use crate::{
+    rcl_bindings::{rcutils_logging_set_default_logger_level, rcutils_logging_set_logger_level},
+    LogParams, LogSeverity, LoggerName, RclrsError, ToLogParams, ToResult, ENTITY_LIFECYCLE_MUTEX,
+};
+
+/// Logger can be passed in as the first argument into one of the [logging][1]
+/// macros provided by rclrs. When passing it into one of the logging macros,
+/// you can optionally apply any of the methods from [`ToLogParams`] to tweak
+/// the logging behavior.
+///
+/// You can obtain a Logger in the following ways:
+/// - Calling [`Node::logger`][2] to get the logger of a Node
+/// - Using [`Logger::create_child`] to create a child logger for an existing logger
+/// - Using [`Logger::new`] to create a new logger with any name that you'd like
+/// - Using [`Logger::default()`] to access the global default logger
+///
+/// Note that if you are passing the Logger of a Node into one of the logging macros,
+/// then you can choose to simply pass in `&node` instead of `node.logger()`.
+///
+/// [1]: crate::log
+/// [2]: crate::Node::logger
+pub struct Logger {
+    name: Box<str>,
+    c_name: CString,
+}
```

Comment:
I believe it was mentioned in the WG meeting that it can be fairly useful to be able to clone the Logger so it can be used from outside the node that defined it. In that case should we derive `Clone`?

Reply:
Great point, and I can change `Box<str>` to `Arc<str>` and `CString` to `Arc<CString>` for better efficiency.

Reply:
Added in https://github.com/geoff-imdex/ros2_rust/pull/13 and just waiting to be merged in that PR